herbert_core/utilities/classes/@serializable/serializable.m

```diff
@@ -1,249 +1,353 @@
-            %                   method
-            %
-            %
-            % Returns:
-            % struc -- structure, or structure array, containing the full
-            %          information, necessary to restore the initial object
-            if nargin>1
-                if isnumeric(varargin{1})
-                    recursively = logical(varargin{1});
-                elseif islogical(varargin{1})
-                    recursively = varargin{1};
-                elseif ischar(varargin{1}) && strncmpi(varargin{1},'-r',2)
-                    recursively = true;
-                else
-                    recursively = false;
-                end
-            else
-                recursively = false;
-            end
-            strc = to_bare_struct_(obj,recursively);
-        end
+        % Convert a serializable object or array of objects into a structure
+        % that includes fields with the class name and version, together with
+        % either:
+        % - the fields of the object (if scalar object)
+        % - a field called array_dat which is an array structure (if object array)
+        S = to_struct (obj)
 
-        %------------------------------------------------------------------
-        function obj = from_bare_struct(obj,inputs)
-            % restore object or array of objects from a plain structure,
-            % previously obtained by to_bare_struct operation
-            % Inputs:
-            % obj    -- non-initialized instance of the object to build
-            % inputs -- the structure, obtained by to_bare_struct method,
-            %           and used as initialization for the object.
-            %
-            % The method takes the list of properties returned by saveableFields
-            % method of the class, and picks up the fields from the
-            % input structure if the fields with these names are present in
-            % the structure.
-            % Then it sets up the values of these fields as the values of
-            % the class properties using the public setters of the input
-            % class.
-            %
-            obj = from_bare_struct_(obj,inputs);
-        end
+        % Restore object or object array from a structure created by to_bare_struct
+        % The input object defines the object class to be recovered, and any
+        % missing fields if recovering from an incomplete structure.
+        obj = from_bare_struct (obj_template, S)
+    end
 
-        function ser_data = serialize(obj)
-            struc = to_struct(obj);
-            ser_data = serialise(struc);
-        end
+    methods (Static)
+        % Restore object or object array from a structure created by to_struct.
+        % The optional input object over-rides the object class type held in the
+        % structure, and provides any missing fields if recovering from an
+        % incomplete structure.
+        obj = from_struct (S, obj_template)
+    end
 
```

Comment:
I would write varargin here. The form makes you think that `obj_template` is a mandatory property while it is rarely used optimization property